pkg/pillar/workerpool/workerpool.go

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// workerpool provides a dynamic set of workers so that work can be
+// spawned without head-of-line blocking
+
+package workerpool
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/agentlog"
+	"github.com/lf-edge/eve/pkg/pillar/worker"
+)
+
+// WorkerPool captures the workers in the pool
+type WorkerPool struct {
+	// Private
+	maxWorkers     int
+	maxWorkersUsed int
+	workers        []myworker
+	numChan        int // Number of result channels from workers
+	resultChan     chan worker.Processor
+	log            worker.Logger
+	ctx            interface{}
+	handlers       map[string]worker.Handler
+	stopTimer      chan struct{}
+}
+
+type myworker struct {
+	worker   *worker.Worker
+	lastUsed time.Time // Last successful submit
+	// XXX do we need to track pending work to avoid deleting while in use?
+	// Or check some queuelen??
+}
+
+// NewWorkerPool constructs a pool
+// If maxWorkers is set to zero it means unlimited
+func NewWorkerPool(log worker.Logger, ctx interface{}, maxWorkers int, handlers map[string]worker.Handler) *WorkerPool {
+	workers := make([]myworker, 0, 10)
```

Comment:
Should this slice capacity be set by `maxWorkers`?

Reply:
No, since it is merely an initial size allocation for the array/slice. Will remove it instead.

Reply:
Done @deitch